scripts/components/map-layers.component.js

```diff
@@ -0,0 +1,169 @@
+import 'styles/components/map/map-layers.scss';
+import 'styles/components/shared/radio-btn.scss';
+import 'styles/components/shared/switcher.scss';
+
+export default class {
+
+  onCreated() {
+    this._setVars();
+    this._setEventListeners();
+  }
+
+  selectedVectorLayers(layers) {
+    this.state = {
```

Comment:
This is a violation of the 'unique source of truth' idea (basically do not duplicate information in the component internal state while it is already in the store). I don't think you even need this `state` obect, since you get the information from the store anyway and it's just passed around inside the component.
LMK if it's not clear I'll try to code it.